### PR TITLE
Bump t8code 3.0.0

### PR DESCRIPTION
**Pull request by jmark**

### CODE REVIEW

**Comment by JoshuaLampert** [comment]
Please request my review once this is finished from your point of view @jmark.

**Comment by jmark** [comment]
All green. The PR is ready for review.

@JoshuaLampert Thank you for your offer doing the review.

File: src/Libt8.jl [hunk 1]
```diff
@@ -7222,8104 +7243,9425 @@ function t8_element_shape_compare(element_shape1, element_shape2)
...
+| local\\_num\\_elements  | Number of elements on this processor.                                                                                                                                                                                                                                                          |
+| global\\_num\\_elements | Number of elements on all processors.                                                                                                                                                                                                                                                          |
+| profile                 | If not NULL, runtimes and statistics about forest\\_commit are stored here.                                                                                                                                                                                                                    |
+"""
+# struct t8_forest
```
Comment (JoshuaLampert) [inline src/Libt8.jl:7286]: If I understand correctly, you commented this and the following lines by hand? Is that correct and if yes, why was that necessary?

File: src/Libt8.jl [hunk 2]
```diff
@@ -17048,61 +18425,35 @@ const P4EST_ENABLE_VTK_BINARY = 1
...
+const P4EST_LIBS = "   m"
 
 const P4EST_PACKAGE = "p4est"
 
 const P4EST_PACKAGE_BUGREPORT = "p4est@ins.uni-bonn.de"
 
 const P4EST_PACKAGE_NAME = "p4est"
 
-const P4EST_PACKAGE_STRING = "p4est 2.8.5.367-931f"
+const P4EST_PACKAGE_STRING = "p4est 0.0.0"
```
Comment (JoshuaLampert) [inline src/Libt8.jl:18442]: Is that how it's supposed to be?

**Review by JoshuaLampert — COMMENTED** [review]
This looks mostly good to me. I have only two questions.

File: src/Libt8.jl (hunk 1 above)
Comment (jmark) [reply src/Libt8.jl:7286]: There is a circular dependency with `t8_forest_t` which throws an error. Since this struct is not supposed to be read and modified directly anyway, I commented this section out and made the struct opaque. I added a comment for clarification.

Comment (jmark) [reply src/Libt8.jl:7286]: This is quick-fix/workaround. Since t8code and Julia is changing so fast I do not see any worth in spending time into a long term solution.

File: src/Libt8.jl (hunk 2 above)
Comment (jmark) [reply src/Libt8.jl:18445]: Of course not. This seems to be an problem in the CMake build system upstream and should be fixed there. The definition in the header files says indeed "0.0.0". Since p4est is a helper library in not supposed to be used by the users directly and this variable is used nowhere, I suggest to leave it that way.

Comment (jmark) [reply src/Libt8.jl:18445]: I opened an issue about that: https://github.com/DLR-AMR/t8code/issues/1294

File: src/Libt8.jl (hunk 1 above)
Comment (JoshuaLampert) [reply src/Libt8.jl:7286]: Hm, this is not ideal since now this file is not the same as you get when you run the generator in `dev/`. Would it be hard to add this patch to `dev/fixes.sh`? Otherwise I see a good chance that we will miss this the next time we run the generator again.

File: src/Libt8.jl (hunk 2 above)
Comment (JoshuaLampert) [reply src/Libt8.jl:18445]: Ok.

File: src/Libt8.jl (hunk 1 above)
Comment (jmark) [reply src/Libt8.jl:7286]: I know and agree. I added an output in `fixes.sh` for the user to apply the fix manually. I won't invest time to cook up a brittle automatic solution which most probably will break and/or gets obsolete in the near future anyway.

Relying on a `fixes.sh` is a nuisance anyway ...

Comment (JoshuaLampert) [reply src/Libt8.jl:7286]: Yeah, that's true. Thanks! Having in mind that you need this PR to be merged to continue working on the other PR(s) in Trixi.jl, I'm fine with the current workaround. But it would be nice to find a proper fix in the future.
